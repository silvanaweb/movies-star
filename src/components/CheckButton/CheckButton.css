.CheckButton {
  background-color: var(--light-grey);
  border-radius: 26px;
  border: none;
  color: var(--teal);
  font-size: 1.6rem;
  padding: var(--s-size) var(--l-size);
}

.CheckButton--checked {
  color: var(--light-grey);
  background-color: var(--teal);
}
