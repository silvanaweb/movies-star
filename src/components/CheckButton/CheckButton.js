import React from 'react';
import './CheckButton.css';

// check button is used for the genre filtering
export const CheckButton = ({ checked, className = '', name, onChange, title }) => {
	const onHandleChange = e => {
		e.stopPropagation();
		onChange({ name, checked: !checked });
	};
	const ButtonCheckedClass = checked ? 'CheckButton--checked' : '';
	return (
		<button className={`CheckButton ${ButtonCheckedClass} ${className}`} onClick={onHandleChange}>
			{title}
		</button>
	);
};
