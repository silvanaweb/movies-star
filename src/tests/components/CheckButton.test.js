import React from "react";
import toJSON from "enzyme-to-json";
import { shallow } from "enzyme";
import { CheckButton } from "../../components/CheckButton/CheckButton";

let onChange, wrapper;
const name = "Action";
const checked = false;

beforeEach(() => {
    onChange = jest.fn();
    wrapper = shallow(
        <CheckButton
            checked={checked}
            className=""
            name={name}
            onChange={onChange}
            title="Action"
        />
    );
});

test("should render CheckButton correctly", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
});

test('should handle onChange when CheckButton is selected', () => {
    wrapper.find('button').simulate('click', {
        stopPropagation: () => { }
    });
    expect(onChange).toHaveBeenLastCalledWith({
        name,
        checked: !checked
    });
});