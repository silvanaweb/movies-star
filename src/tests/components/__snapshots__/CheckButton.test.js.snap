// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CheckButton correctly 1`] = `
<button
  className="CheckButton  "
  onClick={[Function]}
>
  Action
</button>
`;
